src/server/zset_family.cc

```diff
@@ -2279,6 +2279,51 @@ void ZSetFamily::ZRem(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ZSetFamily::ZRandMember(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 0);
+
+  ZRangeSpec range_spec;
+  range_spec.interval = IndexInterval(0, -1);
+  int count = 1;
+
+  switch (args.size()) {
+    case 3: {
+      ToUpper(&args[2]);
+      string_view arg = ArgS(args, 2);
+      range_spec.params.with_scores = arg == "WITHSCORES";
+      if (!range_spec.params.with_scores) {
+        return cntx->reply_builder()->SendError(absl::StrCat("unsupported option ", arg));
+      }
+      [[fallthrough]];
+    }
+    case 2:
+      if (string_view arg = ArgS(args, 1); !absl::SimpleAtoi(arg, &count)) {
+        cntx->reply_builder()->SendError(kInvalidIntErr);
+        return;
+      }
+      break;
+    case 1:
+      break;
+    default:
+      return cntx->reply_builder()->SendError(absl::StrCat("unsupported option ", ArgS(args, 3)));
+  }
+  bool sign = count < 0;
+  range_spec.params.limit = std::abs(count);
```

Comment:
1. I don't think it's simpler than a if (args.size() > x) twice 😅
2. I'll use the chance to advocate for the argparsing library I introduced 🤓 (no need to use, just advertisement)

```c++
#include "facade/cmd_arg_parser.h"

CmdArgParser parser {args};
string_view key = parser.Next();

if (parser.HasNext())
  count = parser.Next().Int<size_t>();

with_scores = parser.Check("WITHSCORES").IgnoreCase()

if (parser.HasNext())
  return (*cntx)->SendError("Unsupported option:" + parser.Next());

if (auto err = parser.Error(); err)
    return (*cntx)->SendError(err->MakeReply());

```

Reply:
It definitely looks better